example/echo_event_loop.ml

```diff
@@ -0,0 +1,172 @@
+open Core
+open Core.Unix
+
+module Ring = struct
+  type t = { ring : (int -> int -> unit) Io_uring.t }
+
+  let make ~ring_size =
+    { ring =
+        Io_uring.create
+          ~max_submission_entries:ring_size
+          ~max_completion_entries:(ring_size * 2)
+    }
+  ;;
+
+  let wait t =
+    let _submitted = Io_uring.submit t.ring in
+    Stdio.Out_channel.flush Stdio.Out_channel.stdout;
+    Io_uring.wait t.ring ~timeout:`Never;
+    Io_uring.iter_completions t.ring ~f:(fun ~user_data ~res ~flags ->
+        user_data res flags);
+    Io_uring.clear_completions t.ring;
+    ()
+  ;;
+
+  let global = make ~ring_size:1024
+end
+
+module Deferred = struct
+  type 'a t =
+    { begin_ : unit -> unit
+    ; mutable then_ : ([ `Ok of 'a | `Eof | `Error ] -> unit) option
+    }
+
+  let create creater_fun =
+    let rec new_t =
+      { begin_ =
+          (fun () ->
+            creater_fun (fun result ->
+                match new_t.then_ with
+                | Some v -> v result
+                | None -> printf "Warning: Ignored Result\n"))
+      ; then_ = None
+      }
+    in
+    new_t
+  ;;
+
+  let bind t f =
+    t.then_ <- Some f;
+    t
```

Comment:
This implementation of `bind` won't work if you bind on the same `t` multiple times, is this intentional?